docs/faqs.md

```diff
@@ -438,3 +438,23 @@ Generally no, these sorts of options can most likely be set from a profile or do
 and would otherwise dynamically embed this into the CMake config files or generated pkg-config files then it should be allowed.
 
 Doing so requires [deleting the option from the `package_id`](adding_packages/conanfile_attributes.md#removing-from-package_id).
+
+## Can I use full_package_mode for a requirement in my recipe?
+
+For some unusual projects, they need to be aligned and being used as requirement, using the very same version, options and settings.
+Those projects usually break between patch versions and are very sentsitive, so we can not use different versions throught Conan graph dependencies,
+otherwise, it may result on unexpected behavior, or even runtime errors.
+
+A very kwnon project is [glib](https://conan.io/center/glib), which requires the very same configuration to prevent multiple instances when using static link. As solution, we could consume glib on full package id mode, like:
+
+```python
+def package_id(self):
+    self.info.requires["glib"].full_package_mode()
+```
+
+Perfect solution on consumer side, but there is a painful side-effect: CCI will not re-generate all involved packages for any change in the dependencies graph which glib is associated, which means, users will start to see **MISSING_PACKAGES** error during their pull requests.
```

Comment:
I am concerned the affect 2.0 will have since this is what the embedded mode will do by default

Reply:
A sheperds topic then

Reply:
I suppose in v2 embed mode should be always active, for C++ and C projects.
as there could be some inline functions in headers, or just macro definitions like:
```
#define SUM(A, B) ((A) + (B))
```
the code from the header is always embedded into the consumer (regardless if it's shared or static library), therefore, if that header changes, all the consumers have to be rebuilt. otherwise, an incompatibility might be encountered (different components use different versions of macro, and such a macro, may define something important for API, like a structure alignment computation).

Reply:
It's the new default depending on package type :)

Reply:
it should be regardless of `package_type`. perhaps, the only exception is header-only depending on header-only.